### PR TITLE
Added $config as first parameter to getList() methods PHPdoc

### LINKED ISSUE #11071: [Bug]: Pimcore\Model\DataObject PHPdoc getList() wrong
### Expected behavior

The getList()-Method of Pimcore\Model\DataObject\Concrete accepts an array as config as the first parameter.
The PHPdoc for this method in generated DataObject classes (Pimcore\Model\DataObject\*) states that getList() does not accept any parameter. This leads to code completion errors in the IDE.

PHPdoc should be

@method static \Pimcore\Model\DataObject\*\Listing getList($listConfig)

### Actual behavior

PHPdoc is

@method static \Pimcore\Model\DataObject\*\Listing getList()

in generated DataObject classes.

### Steps to reproduce

Let pimcore generate a DataObject class and look for @method static \Pimcore\Model\DataObject\*\Listing getList()

### PR DESCRIPTION
**Pull request by leuchtdiode**
Resolves #11071

### CODE REVIEW

File: models/DataObject/ClassDefinition.php [hunk 1]
```diff
@@ -503,7 +503,7 @@ public function generateClassFiles($generateDefinitionFile = true)
         $cd .= 'use Pimcore\Model\DataObject\PreGetValueHookInterface;';
         $cd .= "\n\n";
         $cd .= "/**\n";
-        $cd .= '* @method static \\Pimcore\\Model\\DataObject\\'.ucfirst($this->getName()).'\Listing getList()'."\n";
+        $cd .= '* @method static \\Pimcore\\Model\\DataObject\\'.ucfirst($this->getName()).'\Listing getList($config)'."\n";
```
Comment (blankse) [inline models/DataObject/ClassDefinition.php]:
```suggestion
        $cd .= '* @method static \\Pimcore\\Model\\DataObject\\'.ucfirst($this->getName()).'\Listing getList(array $config = [])'."\n";
```

**Comment by dvesh3** [comment]
@leuchtdiode Thanks for your PR! please sign the CLA for both users `Alex` `leuchtdiode`.

Also, rebase the PR on 10.2 branch, since this is a fix. please see guidelines here https://github.com/pimcore/pimcore/blob/10.x/CONTRIBUTING.md

**Comment by leuchtdiode** [comment]
Idk if I'm doing something wrong, that's my first PR ;) But when I'm trying to rebase onto 10.2, I get a lot of conflicts.

Can I just redo the changes and commit in 10.2 and submit a new PR?

**Comment by leuchtdiode** [comment]
I cherry picked the latest commit from @blankse to 10.2 and created another PR #11087. Is that OK?

**Comment by brusch** [comment]
Closed in favor of #11087

**Pull request closed without merging**